### PR TITLE
only cache logCertSecret namespace secret

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/stolostron/backlog/issues/22048
Signed-off-by: haoqing0110 <qhao@redhat.com>

Add filteredcache.Selector to only cache logCertSecret namespace secret.

Memory usage when creating 500 managedclusters.
```
➜  ~ go tool pprof  http://pprof.apps.server-foundation-410-rbt5h.dev04.red-chesterfield.com/debug/pprof/heap

Fetching profile over HTTP from http://pprof.apps.server-foundation-410-rbt5h.dev04.red-chesterfield.com/debug/pprof/heap

Saved profile in /Users/haoqing/pprof/pprof.controller.alloc_objects.alloc_space.inuse_objects.inuse_space.026.pb.gz
File: controller
Type: inuse_space
Time: May 10, 2022 at 10:30am (CST)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top
Showing nodes accounting for 38.21MB, 84.38% of 45.28MB total
Showing top 10 nodes out of 145
      flat  flat%   sum%        cum   cum%
   12.01MB 26.53% 26.53%    12.01MB 26.53%  reflect.unsafe_NewArray
      10MB 22.09% 48.61%       10MB 22.09%  sigs.k8s.io/json/internal/golang/encoding/json.(*decodeState).literalStore
       3MB  6.63% 55.24%        3MB  6.63%  reflect.New
    2.56MB  5.66% 60.90%     2.56MB  5.66%  k8s.io/apimachinery/pkg/watch.(*Broadcaster).Watch.func1
    2.50MB  5.52% 66.42%     2.50MB  5.52%  reflect.mapassign
       2MB  4.42% 70.84%        2MB  4.42%  k8s.io/apimachinery/pkg/runtime.(*RawExtension).UnmarshalJSON
    1.57MB  3.47% 74.32%     2.59MB  5.73%  k8s.io/apimachinery/pkg/runtime.(*Scheme).AddKnownTypeWithName
    1.55MB  3.42% 77.74%     1.55MB  3.42%  k8s.io/apimachinery/pkg/watch.NewLongQueueBroadcaster
    1.51MB  3.33% 81.06%     1.51MB  3.33%  github.com/stolostron/multicloud-operators-foundation/pkg/utils.Mergesubjects
    1.50MB  3.32% 84.38%     1.50MB  3.32%  k8s.io/apimachinery/pkg/api/meta.(*DefaultRESTMapper).AddSpecific
```
Screen Shot 2022-05-10 at 10 32 53

### CODE REVIEW

File: cmd/controller/app/server.go [hunk 1]
```diff
@@ -98,11 +103,32 @@ func Run(o *options.ControllerRunOptions, ctx context.Context) error {
 	clusterInformers := clusterv1informers.NewSharedInformerFactory(clusterClient, 10*time.Minute)
 	kubeInfomers := kubeinformers.NewSharedInformerFactory(kubeClient, 10*time.Minute)
 
+	// Cache only the LogCertSecret namespace secret
```
Comment (qiujian16) [inline cmd/controller/app/server.go:106]: can we have more comments here? does it mean if we want to watch other secret in the future, we will need to update the filter here?

Comment (haoqing0110) [reply cmd/controller/app/server.go]: @qiujian16 yes, add more comments here.

**Comment by qiujian16** [comment]
/approve

@elgnay or @zhiweiyin318 PTAL

**Pull request merged**